### PR TITLE
Update ZE to API v1.0

### PR DESCRIPTION
**Pull request by zhenggb72**
## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Review by hzhou — CHANGES_REQUESTED** [review]
@zhenggb72 Could you split the commit "mpl/gpu: update to new Level zero API v1.0" into two commits, one for mpl/gpu, and the other for test/mpi?

Could you add commit message highlight the major changes?

As long as the commits are well organized and as long as it doesn't affect other code, I guess we can quickly merge this PR.

**Comment by zhenggb72** [comment]
> @zhenggb72 Could you split the commit "mpl/gpu: update to new Level zero API v1.0" into two commits, one for mpl/gpu, and the other for test/mpi?
> 

Done

> Could you add commit message highlight the major changes?
> 
> As long as the commits are well organized and as long as it doesn't affect other code, I guess we can quickly merge this PR.

Level-zero v1.0 is a major update and it is not compatible with previous versions. Basically many function names , data structures,  have been re-defined.

File: src/mpl/src/gpu/mpl_gpu_ze.c [hunk 1]
```diff
@@ -11,8 +11,10 @@ MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
 #ifdef MPL_HAVE_ZE
 
 ze_driver_handle_t global_ze_driver_handle;
-ze_device_handle_t *global_ze_devices_handle;
-int gpu_ze_init_driver();
+ze_device_handle_t *global_ze_devices_handle = NULL;
```
Comment (hzhou) [inline src/mpl/src/gpu/mpl_gpu_ze.c:14]: Can we add a comment at the beginning of this file, clarifying that we are supporting level zero API v1.0? Is there a version check API to confirm that we are using the correct ze library?

File: test/mpi/include/mtest_common.h [hunk 2]
```diff
@@ -16,6 +16,7 @@ typedef enum {
     MTEST_MEM_TYPE__UNREGISTERED_HOST,
     MTEST_MEM_TYPE__REGISTERED_HOST,
     MTEST_MEM_TYPE__DEVICE,
+    MTEST_MEM_TYPE__SHARED,
```
Comment (hzhou) [inline test/mpi/include/mtest_common.h:19]: Is ze "shared" memory similar to cuda "managed" memory?

File: src/mpl/src/gpu/mpl_gpu_ze.c (hunk 1 above)
Comment (zhenggb72) [reply src/mpl/src/gpu/mpl_gpu_ze.c:14]: Level zero API is still evolving. At this time, we will always support the latest version and not maintaining the backward compatibility.  In v1.0, even the VERSION string macro changed into a runtime checking (via a function new function API), so there is no reliable/stable version check API at this time.

File: test/mpi/include/mtest_common.h (hunk 2 above)
Comment (zhenggb72) [reply test/mpi/include/mtest_common.h:19]: Yes. they are similar.

File: src/mpl/src/gpu/mpl_gpu_ze.c (hunk 1 above)
Comment (hzhou) [reply src/mpl/src/gpu/mpl_gpu_ze.c:14]: I see. That's fine. Can we at least add a comment (at the beginning of this file) on the API version and a known ze library build number/so version (there must be something that we can use to identify the library, right?)?

Comment (zhenggb72) [reply src/mpl/src/gpu/mpl_gpu_ze.c:17]: done

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/most

**Review by hzhou — APPROVED** [review]
Approved. Just rebase and we'll merge it.

**Comment by zhenggb72** [comment]
rebased

**Pull request merged**